packages/babel-types/src/definitions/es2015.js

```diff
@@ -100,21 +134,25 @@ defineType("ClassDeclaration", {
   ],
   aliases: ["Scopable", "Class", "Statement", "Declaration", "Pureish"],
   fields: {
-    id: {
-      validate: assertNodeType("Identifier"),
-    },
-    body: {
-      validate: assertNodeType("ClassBody"),
+    ...classCommon,
+    declare: {
+      validate: assertValueType("boolean"),
+      optional: true,
     },
-    superClass: {
+    abstract: {
+      validate: assertValueType("boolean"),
       optional: true,
-      validate: assertNodeType("Expression"),
+    },
+    id: {
+      validate: assertNodeType("Identifier"),
+      optional: true, // Missing if this is the child of an ExportDefaultDeclaration.
```

Comment:
Same reason as `function`. They're both special-cased in the spec to allow omitting the name specifically in this case.